aten/src/ATen/native/TensorConversions.cpp

```diff
@@ -19,13 +19,32 @@ static inline Device ensure_has_index(Device device) {
   return impl->getDevice();
 }
 
-static inline Tensor to_impl(const Tensor& self, const TensorOptions& options, bool non_blocking) {
-  auto r = at::empty(self.sizes(), options);
+static inline Tensor to_impl(const Tensor& self, const TensorOptions& options, bool non_blocking, bool copy, c10::optional<c10::MemoryFormat> optional_memory_format) {
+  auto memory_format =
+      optional_memory_format.value_or(MemoryFormat::Contiguous);
+
+  if (self.options() == options && !copy &&
```

Comment:
Don't compare options. See https://github.com/pytorch/pytorch/issues/25478 and  https://github.com/pytorch/pytorch/pull/27572 (btw, bikeshedding on TensorAxes welcome)

Reply:
Result of in person conversation: Let's add something like "match TensorOptions with Tensor". This tests if a Tensor is "compatible" with a TensorOptions. (This is not an equality test, because options can be left unspecified, and in that case you always match.)